src/ContextualLogger.ts

```diff
@@ -0,0 +1,77 @@
+import { IContext, IJSLoggerDefaultsType, ILogger, ILoggerOpts, ILogLevel, JSLoggerExportType, LogHandler } from "./datatypes";
+export * from "./datatypes";
+import { JSLoggerDefaults as Logger } from "./JSLoggerDefaults";
+import { bind, defineLogLevel } from "./utils";
+
+// Inner class which performs the bulk of the work; ContextualLogger instances can be configured independently
+// of each other.
+export class ContextualLogger implements ILogger {
+  public readonly context: IContext;
```

Comment:
Any reason for making the context public?

Reply:
The global logger's context is used when creating a new logger in https://github.com/tp/js-logger/blob/077cea9477c2c3a225d301dc98190c67994fe413/src/JSLoggerDefaults.ts#L46:L52